### PR TITLE
Feature create lib

### PR DESCRIPTION
**Pull request by petterarvidsson**

### CODE REVIEW

**Comment by petterarvidsson** [comment]
@nsg as discussed, let's split the project into the main application (which maybe we should also split into several files) and the library that contains all code to handle the different parts required by main. What do you think?

File: lib/src/matrix.cpp [hunk 1]
```diff
@@ -57,7 +55,7 @@ namespace konstructs {
             float ymax, xmax;
             ymax = znear * tanf(fov * PI / 360.0);
             xmax = ymax * aspect;
-            return frustum(-xmax, xmax, -ymax, ymax, znear, zfar);
+            return projection_frustum(-xmax, xmax, -ymax, ymax, znear, zfar);
```
Comment (petterarvidsson) [inline lib/src/matrix.cpp]: @nsg Apparently we have been using the frustum function from nanogui without knowing it... the end result is identical with our own version.

Comment (nsg) [inline lib/src/matrix.cpp]: Oh! :)

**Comment by nsg** [comment]
@petterarvidsson looks good!

**Comment by petterarvidsson** [comment]
@nsg Git is really awesome. I rebased this on master and even after all our changes to the client code, merge was automatic! Shall we merge?

**Comment by nsg** [comment]
@petterarvidsson :+1:

**Pull request merged**